### PR TITLE
Allow `agent integration install` to work when the configuration file is missing

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Ignore missing configuration file if:
— the configuration file is the default one and has not been explicitly set;
— the command run is `agent integration install`.

### Motivation

In order to add custom checks to a datadog agent docker image, the procedure involves running `agent integration install` inside a Dockerfile: [Community integration installation](https://docs.datadoghq.com/agent/guide/community-integrations-installation-with-docker-agent/?tab=docker).
The issue is that the datadog agent docker image doesn't have a `/etc/datadog-agent/datadog.yaml` file. Instead, it dynamically creates it at Kubernetes pod start: https://github.com/DataDog/datadog-agent/blob/fc7b205536a8f5fa16d814ed934ad76dfe4816e8/Dockerfiles/agent/entrypoint/51-docker.sh#L16-L19

As this init script isn’t run in the context of a `docker build` used to build a custom image, `agent integration install` fails.

### Additional Notes

As a short-term workaround, the doc has been amended to add an explicit specification of a datadog configuration file in this context: https://github.com/DataDog/documentation/pull/7208

This workaround can now be reverted: https://github.com/DataDog/documentation/pull/8480

### Describe your test plan

Try to build a custom image with this Dockerfile:
```dockerfile
FROM python:3.8 AS wheel_builder
WORKDIR /wheels
RUN pip install "datadog-checks-dev[cli]"
RUN git clone https://github.com/DataDog/integrations-extras.git
RUN ddev config set extras ./integrations-extras
RUN ddev -e release build bind9

FROM datadog/agent:latest
COPY --from=wheel_builder /wheels/integrations-extras/bind9/dist/ /dist
RUN agent integration install -r -w /dist/*.whl
```

With `FROM datadog/agent:7.22.0`, we got:
```
Step 9/9 : RUN agent integration install -r -w /dist/*.whl
 ---> Running in 58acaac2acdd
Cannot setup config, exiting: unable to load Datadog config file: Config File "datadog" Not Found in "[/etc/datadog-agent]"
Error: unable to load Datadog config file: Config File "datadog" Not Found in "[/etc/datadog-agent]"
The command '/bin/sh -c agent integration install -r -w /dist/*.whl' returned a non-zero code: 255
```

With `FROM datadog/agent:7.23.0-rc.1`, we should get:
```
Step 9/9 : RUN agent integration install -r -w /dist/*.whl
 ---> Running in 1d1e2e2fb374
For your security, only use this to install wheels containing an Agent integration and coming from a known source. The Agent cannot perform any verification on local wheels.
Processing /dist/datadog_bind9-1.0.0-py2.py3-none-any.whl
Installing collected packages: datadog-bind9
Successfully installed datadog-bind9-1.0.0
Successfully copied configuration file conf.yaml.example
Successfully completed the installation of datadog-bind9
Removing intermediate container 1d1e2e2fb374
 ---> db02d569ec3d
Successfully built db02d569ec3d
```

### CODE REVIEW

File: cmd/agent/common/helpers.go [hunk 1]
```diff
@@ -53,8 +63,10 @@ func setupConfig(confFilePath string, configName string, withoutSecrets bool) (*
 	} else {
 		warnings, err = config.Load()
 	}
-	if err != nil {
-		return warnings, fmt.Errorf("unable to load Datadog config file: %s", err)
+	// If `!failOnMissingFile`, do not issue an error if we cannot find the default config file.
+	var e viper.ConfigFileNotFoundError
+	if err != nil && !(!failOnMissingFile && errors.As(err, &e) && confFilePath == "") {
```
Comment (prognant) [inline cmd/agent/common/helpers.go:68]: 👍

**Review by prognant — APPROVED** [review]
LGTM

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]
Looks fine to me, doesn't seem to change how `datadog-checks-downloader` works or is used

Also, you might want to use `agent integration install --extra datadog-bind9==1.0.0` now instead of building it from source...

Cc @FlorianVeaux @ofek

File: cmd/agent/common/helpers.go (hunk 1 above)
Comment (L3n41c) [reply cmd/agent/common/helpers.go]: Done !
The `if` condition is now in normal conjonctive form. ;-)

**Comment by L3n41c** [comment]
> Also, you might want to use agent integration install --extra datadog-bind9==1.0.0 now instead of building it from source...

Should we update the official doc as well: [Community integration installation](https://docs.datadoghq.com/agent/guide/community-integrations-installation-with-docker-agent/?tab=docker) ?
https://github.com/DataDog/documentation/blob/f3405f3bf842f9f8338ebb62cff5fae455fdc95f/content/en/agent/guide/community-integrations-installation-with-docker-agent.md#L71

File: cmd/agent/common/helpers.go [hunk 2]
```diff
@@ -20,16 +22,24 @@ func SetupConfig(confFilePath string) error {
 
 // SetupConfigWithWarnings fires up the configuration system and returns warnings if any.
 func SetupConfigWithWarnings(confFilePath, configName string) (*config.Warnings, error) {
```
Comment (trishankatdatadog) [inline cmd/agent/common/helpers.go:24]: Slightly off-topic, but isn't it more appropriate to call `SetupConfigWithWarnings` `SetupConfigWithSecrets` instead?

**Review by trishankatdatadog — APPROVED** [review]
LGTM from a security point of view, does not affect datadog-checks-downloader security.

**Pull request merged**